poly1305/src/backend/avx2.rs

```diff
@@ -196,4 +196,8 @@ impl UhfBackend for State {
             }
         }
     }
+
+    fn blocks_needed_to_align(&self) -> usize {
+        self.cached_blocks.len() - self.num_cached_blocks
```

Comment:
If `self.num_cached_blocks = 0` then this will return `4`, when it should actually return 0. In fact, this function will _never_ return 0, because `self.num_cached_blocks` is never set to 4. This means that the fast path will never be taken.

Instead, make this:
```suggestion
        if self.num_cached_blocks == 0 {
            // There are no cached blocks; fast path is available.
            0
        } else {
            // There are cached blocks; report how many more we need.
            self.cached_blocks.len() - self.num_cached_blocks
        }
```

Reply:
Good catch, thanks!